doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+```
+return (<expression>)?
+
+if a > b:
+  return a
+else:
+  return b
+end
+```
+
+### Control structures
+
+To control the flow of executioon, NESTML supports loops and conditionals.
+
+#### Loops
+
+The start of the `while` loop is composed of the keyword `while` followed by a boolean condition and a colon. It is closed with the keyword `end`. It executes the statements inside the block as long as the given boolean expression evaluates to `true`.
+```
+while <boolean_expression>:
+  <statements>
+end
+
+x integer = 0
+while x <= 10:
+  y = max(3, x)
+end
+```
+
+The `for` loop starts with the keyword `for` followed by the name of a previously defined variable of type `integer` or `real`. The fist variant uses an `integer` stepper variable which iterates over the half-open interval [`lower_bound`, `upper_bound`) in steps of 1.
+```
+for <existing_variable_name> in <lower_bound> ... <upper_bound>:
+  <statements>
+end
+
+x integer = 0
+for x in 1 ... 5:
+  <statements>
+end
+```
+The second variant uses an `integer` or `real` iterator variable and iterates over the half-open interval `[lower_bound, upper_bound)` with a positive `integer` or `real` step of size `step`. It is advisable to chose the type of the iterator variable and the step size to be the same.
+```
+for <existing_variable_name> in <lower_bound> ... <upper_bound> step <step>:
+  <statements>
+end
+
+x integer
+for x in 1 ... 5 step 2:
+  <statements>
+end
+
+x real
+for x in 0.1 ... 0.5 step 0.1:
+  Statements
+end
+```
+
+#### Conditionals
+
+NESTML supports different variants of the if-else conditional. The first example shows the `if` conditional composed of a single `if` block:
+
+```
+if <boolean_expression>:
+  <statements>
+end
+
+if 2 < 3:
+  <statements>
+end
+```
+
+The second example shows an if-else block, which executes the `if_statements` in case the boolean expression evaluates to true and the `else_statements` else.
+```
+if <boolean_expression>:
+  <if_statements>
+else:
+  <else_statements>
+end
+
+if 2 < 3:
+  <if_statements>
+else:
+  <else_statements>
+end
+```
+
+In order to allow grouping a sequence of related `if` conditions, NESTML also supports the `elif`-conditionals. An `if` condition can be followed by an arbitrary number of `elif` conditions. Optionally, this variant also supports the `else` keyword for a catch-all statement. The whole conditional always concludes with an `end` keyword.
+```
+if <boolean_expression>:
+  <if_statements>
+elif <boolean_expression>:
+  <elif_statements>
+else:
+  <else_statements>
+end
+
+if 2 < 3:
+  <if_statements>
+elif 4>6:
+  <elif_statements>
+else:
+  <else_statements>
+end
+
+if 2 < 3:
+  <if_statements>
+elif 4>6:
+  <elif_statements>
+end
+```
+
+Conditionals can also be nested inside of each other.
+```
+if 1 < 4:
+  <statements>
+  if 2 < 3:
+    <statements>
+  end
+end
+```
+
+### Expressions and operators
+
+Expressions in NESTML can be specified in a recursive fashion.
```

Comment:
@DimitriPlotnikov: this sentence is a bit lonely here. What does it mean and why is this relevant to the user? Can you please elaborate a bit on this?